### ISSUE TITLE
coord_sf() doesn't adjust line size for graticule lines

### ISSUE THREAD

**Issue by clauswilke** [tidyverse/ggplot2#2991]
Line sizes in themes are generally provide in units of mm and converted to grid units by multiplication by `.pt`. However, `coord_sf()` doesn't do that, it uses the value of `size` directly. Reprex:

``` r
library(ggplot2)

df <- data.frame(x = c(1, 2, 3), y = c(1, 2, 3))
p <- ggplot(df, aes(x, y)) + geom_point()

cowplot::plot_grid(
  p + coord_fixed() + theme(panel.grid.minor = element_blank()) +
    ggtitle("coord_fixed()"),
  p + coord_sf() +
    ggtitle("coord_sf() default"),
  p + coord_sf() + theme(panel.grid = element_line(size = 0.5*.pt)) +
    ggtitle("coord_sf() theme adjusted"),
  nrow = 2
)
```

![](https://i.imgur.com/pbhs9MG.png)

<sup>Created on 2018-11-09 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

The offending lines are here:
https://github.com/tidyverse/ggplot2/blob/6786969ecb2ad38876ec1df43969467339d7818e/R/sf.R#L576-L582
The fix is simple, but it would affect the visual appearance of a lot of plots out in the wild.

**Comment by yutannihilation** [comment]
+1

**Comment by clauswilke** [comment]
There's a somewhat related issue that could be fixed at the same time: #2525
